lib/src/standardGeneratedSections.cc

```diff
@@ -119,33 +118,32 @@ void StandardGeneratedSections::neuronCurrentInjection(
     const std::string &ftype,
     const std::string &rng)
 {
-    os << "// pull injected current variables in a coalesced access" << std::endl;
-    const auto css = ng.getCurrentSources();
-    for (const auto *cs : css)
-    {
-        const auto* csm = cs->getCurrentSourceModel();
-        VarNameIterCtx csVars(csm->getVars());
-        // store the defined parts of the neuron state into the global state variables dd_V etc
-        for(const auto &v : csVars.container) {
-            os <<  v.second << " l" << v.first << cs->getName() << " = ";
-            os << devPrefix << v.first << cs->getName() << "[" << localID << "];" << std::endl;
-        }
-    }
-    os << "// add injected current from sources" << std::endl;
-    for (const auto *cs : css)
+    // Loop through all of neuron group's current sources
+    for (const auto *cs : ng.getCurrentSources())
     {
+        os << "// current source " << cs->getName() << std::endl;
+        CodeStream::Scope b(os);
+
         const auto* csm = cs->getCurrentSourceModel();
         VarNameIterCtx csVars(csm->getVars());
         DerivedParamNameIterCtx csDerivedParams(csm->getDerivedParams());
         ExtraGlobalParamNameIterCtx csExtraGlobalParams(csm->getExtraGlobalParams());
-        os << "// current source " << cs->getName() << std::endl;
-        if (!csm->getInjectionCode().empty()){
-            string iCode = csm->getInjectionCode();
-            substitute(iCode, "$(id)", localID);
-            StandardSubstitutions::currentSourceInjection(iCode, cs,
-                                csVars, csDerivedParams, csExtraGlobalParams,
-                                functions, ftype, rng);
-            os << iCode << std::endl;
+
+        // Read current source variables into registers
+        for(const auto &v : csVars.container) {
+            os <<  v.second << " l" << v.first << " = " << devPrefix << v.first << cs->getName() << "[" << localID << "];" << std::endl;
```

Comment:
In theory there can be several current sources per population and if some of their variables have the same name, there will be a collision. For example, if for any reason I want to inject DC into half of the population, and AC into another half, controlling the injection by, say boolean ``applyCurrent`` variable in both cases, then ``applyCurrent`` will be defined twice. Appending name of the current source solves it.

Reply:
This is why I put all the code for each current source in a scope (that's what the ``CodeStream::Scope b(os);`` bit does) - if you look at the generated code it'll be enclosed in {} so the fact the names are the same doesn't matter.

Reply:
ahh my bad! Then this PR is good to go.

Reply:
Do you have a fix for the extra global parameters bug we can cherry pick in?

Reply:
I did not commit it yet, so I think I will simply push it here